plugins/modules/system/sudoers.py

```diff
@@ -0,0 +1,214 @@
+    name:
+        required: true
+        description:
+            - Name of the sudoers rule
+        type: str
+    nopassword:
+        required: false
+        description:
+            - Whether a password will be required to run the sudo'd command
+        default: true
+        type: bool
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the rule should exist or not
+        type: str
+    user:
+        required: false
+        description:
+            - Name of the user for the sudoers rule
+            - (cannot be used in conjunction with group)
+        type: str
+'''
+
+EXAMPLES = '''
+- name: Allow the backup user to sudo /usr/local/bin/backup
+  sudoers:
+    name: allow-backup
+    state: present
+    user: backup
+    command: /usr/local/bin/backup
+
+- name: Allow the monitoring group to run sudo /usr/local/bin/gather-app-metrics without requiring a password
+  sudoers:
+    name: monitor-app
+    group: monitoring
+    command: /usr/local/bin/gather-app-metrics
+
+- name: Allow the alice user to run sudo /bin/systemctl restart my-service or sudo /bin/systemctl reload my-service, but a password is required
+  sudoers:
+    name: alice-service
+    user: alice
+    command:
+      - /bin/systemctl restart my-service
+      - /bin/systemctl reload my-service
+    nopassword: false
+
+- name: Revoke the previous sudo grants given to the alice user
+  sudoers:
+      name: alice-service
+      state: absent
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+
+
+class Sudoers(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.name = module.params['name']
+        self.user = module.params.get('user')
+        self.group = module.params.get('group')
+        self.state = module.params['state']
+        self.nopassword = bool(module.params.get('nopassword', True))
+        self.file = '/etc/sudoers.d/{filename}'.format(filename=self.name)
+
+        command = module.params['command']
+        if isinstance(command, list):
+            self.commands = command
+        else:
+            self.commands = [command]
+
+    def write(self):
+        with open(self.file, 'w') as f:
+            f.write(self.content())
+
+    def delete(self):
+        os.remove(self.file)
+
+    def exists(self):
+        return os.path.exists(self.file)
+
+    def matches(self):
+        with open(self.file, 'r') as f:
+            return f.read() == self.content()
+
+    def content(self):
+        if self.user:
+            owner = self.user
+        elif self.group:
+            owner = '%{group}'.format(group=self.group)
+
+        command_str = ', '.join(self.commands)
+        nopasswd_str = 'NOPASSWD' if self.nopassword else ''
+        return "{owner} ALL={nopasswd}: {command}\n".format(owner=owner, nopasswd=nopasswd_str, command=command_str)
+
+    def check(self):
+        try:
```

Comment:
I think there's some opportunity to improve readability here:
- `run` and `check` should not be separate functions, but instead `run` should add conditionals to avoid execution when `check_mode = True`.
- The `try` block should be in main
- `exit_json` should also be in main and `run` should just return `True/False` which should help avoid nesting

Reply:
Thanks for the suggestion to revisit this!